src/main/java/com/google/sps/servlets/AddSchoolData.java

```diff
@@ -0,0 +1,132 @@
+package com.google.sps.servlets.second;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/add-info")
+public class AddSchoolData extends HttpServlet {
+
+  private Boolean isNewSchool = true;
+  private Boolean isNewCourse = true;
+  private Boolean isNewProfessor = true;
+
+  private Key existingSchoolKey;
+  private Key existingCourseKey;
+  private Key existingProfessorKey;
+
+  private DatastoreService db = DatastoreServiceFactory.getDatastoreService();
+
+  @Override
```

Comment:
Nice job cleaning this up. It feels a lot more readable now, and hopefully it's also easier to test.
Just a few things and then should be good to go.